azure-storage/spring-cloud-starter-stream-sink-azure-blob/src/main/java/org/springframework/cloud/stream/app/azure/storage/sink/AzureBlobSinkConfiguration.java

```
@@ -0,0 +1,140 @@
+            if (this.properties.getPublicPermission()) {
+                LOG.info("getBlobService() : making container publically accessible");
+
+                // Create a permissions object
+                BlobContainerPermissions containerPermissions = new BlobContainerPermissions();
+
+                // Include public access in the permissions object
+                containerPermissions.setPublicAccess(BlobContainerPublicAccessType.CONTAINER);
+
+                // Set the permissions on the container
+                container.uploadPermissions(containerPermissions);
+            }
+
+            LOG.info("getBlobService() : using blob name {}", this.properties.getBlobName());
+            
+            if (this.properties.getAppendOnly()) {
+                this.blobService = container.getAppendBlobReference(this.properties.getBlobName());
+                if (this.properties.getOverwiteExistingAppend()) {
+                    ((CloudAppendBlob) blobService).createOrReplace();
+                }
+            }
+            else {
+                this.blobService = container.getBlockBlobReference(this.properties.getBlobName());
+            }
+        } catch (Exception e) {
+            // Log the stack trace.
+            LOG.error("getBlobService() : {}", e.getMessage());
+            //throw e;
+        }
+    }
+
+    @ServiceActivator(inputChannel=Sink.INPUT)
+    public void pushToAzureBlob(Message<?> message) throws MessagingException {
+        try {
+            // Upload the payload to the blob
+            if (this.properties.getAppendOnly()) {
+                ((CloudAppendBlob) blobService).appendText(message.getPayload().toString() + "\n");
+            }
+            else {
+                ((CloudBlockBlob) blobService).uploadText(message.getPayload().toString());
+            }
+
+        } catch (Exception e) {
+            // Log the stack trace.
+            LOG.error("pushToAzureBlob() : {}", e.getMessage());
```

This solution doesn't look good.
Would be better just don't catch any exceptions and let them bubble to the caller.